src/python/model_api/visualizer/scene/keypoint.py

```diff
@@ -3,19 +3,30 @@
 # Copyright (C) 2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+from typing import Union
+
+from PIL import Image
+
 from model_api.models.result import DetectedKeypoints
 from model_api.visualizer.layout import Flatten, Layout
-from model_api.visualizer.primitive import Overlay
+from model_api.visualizer.primitive import Keypoint
 
 from .scene import Scene
 
 
 class KeypointScene(Scene):
     """Keypoint Scene."""
 
-    def __init__(self, result: DetectedKeypoints) -> None:
-        self.result = result
+    def __init__(self, image: Image, result: DetectedKeypoints, layout: Union[Layout, None] = None) -> None:
+        super().__init__(
+            base=image,
+            keypoints=self._get_keypoints(result),
+            layout=layout,
+        )
+
+    def _get_keypoints(self, result: DetectedKeypoints) -> list[Keypoint]:
```

Comment:
@ashwinvaidya17, would it be an idea to use predictions instead of results? What is results object? A prediction? Or something else?